ampd/src/handlers/evm_verify_msg.rs

```diff
@@ -160,18 +145,20 @@ where
 {
     type Err = Error;
 
-    async fn handle(&self, event: &event_sub::Event) -> Result<()> {
-        let Event {
+    async fn handle(&self, event: &events::Event) -> Result<()> {
+        let PollStartedEvent {
             contract_address,
             poll_id,
             source_chain,
             source_gateway_address,
             messages,
             confirmation_height,
             participants,
-        } = match event.try_into().into_report()? {
-            Some(event) => event,
-            None => return Ok(()),
+        } = match event.try_into() as error_stack::Result<PollStartedEvent, events::Error> {
+            Err(report) if matches!(report.current_context(), EventTypeMismatch(_)) => {
+                return Ok(())
+            }
+            event => event.change_context(DeserializeEvent)?,
```

Comment:
This beomes a bit too much though.

Reply:
that's a side-effect of <error-stack::Report> unfortunately. Haven't found a better option so far